flash/image/instance_segmentation/model.py

```diff
@@ -0,0 +1,91 @@
+# limitations under the License.
+from typing import Any, Dict, List, Mapping, Optional, Type, Union
+
+import torch
+from torch.optim import Optimizer
+
+from flash.core.data.process import Serializer
+from flash.core.integrations.icevision.model import IceVisionTask, SimpleCOCOMetric
+from flash.core.registry import FlashRegistry
+from flash.core.utilities.imports import _ICEVISION_AVAILABLE
+from flash.image.instance_segmentation.backbones import INSTANCE_SEGMENTATION_HEADS
+
+if _ICEVISION_AVAILABLE:
+    from icevision.metrics import COCOMetricType
+    from icevision.metrics import Metric as IceVisionMetric
+
+
+class InstanceSegmentation(IceVisionTask):
+    """The ``InstanceSegmentation`` is a :class:`~flash.Task` for detecting objects in images. For more details, see
+    :ref:`object_detection`.
+
+    Args:
+        num_classes: the number of classes for detection, including background
+        model: a string of :attr`_models`. Defaults to 'fasterrcnn'.
+        backbone: Pretained backbone CNN architecture. Constructs a model with a
+            ResNet-50-FPN backbone when no backbone is specified.
+        fpn: If True, creates a Feature Pyramind Network on top of Resnet based CNNs.
+        pretrained: if true, returns a model pre-trained on COCO train2017
+        pretrained_backbone: if true, returns a model with backbone pre-trained on Imagenet
+        trainable_backbone_layers: number of trainable resnet layers starting from final block.
+            Only applicable for `fasterrcnn`.
+        loss: the function(s) to update the model with. Has no effect for torchvision detection models.
+        metrics: The provided metrics. All metrics here will be logged to progress bar and the respective logger.
+            Changing this argument currently has no effect.
+        optimizer: The optimizer to use for training. Can either be the actual class or the class name.
+        pretrained: Whether the model from torchvision should be loaded with it's pretrained weights.
+            Has no effect for custom models.
+        learning_rate: The learning rate to use for training
+
+    """
+
+    heads: FlashRegistry = INSTANCE_SEGMENTATION_HEADS
+
+    required_extras: str = "image"
+
+    def __init__(
+        self,
+        num_classes: int,
+        backbone: Optional[str] = "resnet18_fpn",
+        head: Optional[str] = "mask_rcnn",
+        pretrained: bool = True,
+        metrics: Optional['IceVisionMetric'] = None,
+        optimizer: Type[Optimizer] = torch.optim.Adam,
+        learning_rate: float = 5e-4,
+        serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+        image_size: Optional[int] = None,
+        **kwargs: Any,
+    ):
+        self.save_hyperparameters()
+
+        super().__init__(
+            num_classes=num_classes,
+            backbone=backbone,
+            head=head,
+            pretrained=pretrained,
+            metrics=metrics or [SimpleCOCOMetric(COCOMetricType.mask)],
```

Comment:
Not sure you need to wrap into a list.

Reply:
This is being passed through to the icevision lightningmodule, which expects a list